### PR TITLE
Fix queue position message edits being recognized  in chatSDK onNewMessage()

### PR DESCRIPTION
**Pull request by kevinlin98z**
**Problem:**
`chat.onNewMessage()` was not recognizing queue position message edits. Investigations traced root cause to queue position message edits coming in as `chatMessageEdited` event type, not `chatMessageReceived`.

**Solution:**
Add a `chatMessageEdited` listener to listen to those queue position message edit events. `chatMessageEdited` events contain this `editedOn` property. For this edited message, `createdOn` timestamp will be updated with `editedOn` timestamp. 

Video demo: https://microsoft-my.sharepoint.com/:v:/p/kevinlin/EZsNY7oI5QRGqlacnUVDbo8BlT8Sy2pQ5CRqq-tdTWOeSw?e=PtIX9k

image
image
image

### CODE REVIEW

File: src/core/messaging/ACSClient.ts [hunk 1]
```diff
@@ -219,7 +219,7 @@ export class ACSConversation {
                 }
 
                 // Filter out duplicate messages
-                if (postedMessageIds.has(id)) {
+                if (postedMessageIds.has(id) && postedMessageIds.has(version)) {
```
Comment (charliewang95) [inline src/core/messaging/ACSClient.ts:222]: Where is `version` being added to `postedMessageIds`? And shouldn't we use `id + version` to identify a unique record inside `postedMessageIds`?

Comment (charliewang95) [inline src/core/messaging/ACSClient.ts:222]: Are we certain `version` will always take the same format as `id`? Is it safer if we use something like `${id}_version_${version}` inside the `postedMessageIds` set?

Comment (kevinlin98z) [reply src/core/messaging/ACSClient.ts]: After discussion with @xTEddie , updated logic to rely on `editedOn` property instead of `version`.

Comment (kevinlin98z) [reply src/core/messaging/ACSClient.ts]: After discussion with @xTEddie , updated logic to rely on `editedOn` property instead of `version`.

**Review by xTEddie — APPROVED** [review]

**Pull request merged**